pkg/node/node.go

```diff
@@ -424,6 +425,53 @@ func (n *Node) publishSignature(
 			continue
 		}
 
+		currentBlock, err := n.ethereumChain.BlockCounter().CurrentBlock()
+		if err != nil {
+			logger.Errorf(
+				"could not get current block while confirming "+
+					"signature for keep [%s]: [%v] ",
+				keepAddress.String(),
+				err,
+			)
+			time.Sleep(retryDelay) // TODO: #413 Replace with backoff.
+			continue
+		}
+
+		isSignatureConfirmed, err := chainutil.WaitForBlockConfirmations(
+			n.ethereumChain.BlockCounter(),
+			currentBlock,
+			blockConfirmations,
+			func() (bool, error) {
+				isAwaitingSignature, err := n.ethereumChain.IsAwaitingSignature(
+					keepAddress,
+					digest,
+				)
+				if err != nil {
+					return false, err
+				}
+
+				return !isAwaitingSignature, nil
+			},
+		)
+		if err != nil {
+			logger.Errorf(
+				"could not confirm signature for keep [%s]: [%v] ",
+				keepAddress.String(),
+				err,
+			)
+			time.Sleep(retryDelay) // TODO: #413 Replace with backoff.
+			continue
+		}
+
+		if !isSignatureConfirmed {
+			logger.Errorf(
+				"signature for keep [%s] is not confirmed",
```

Comment:
`"signature submission for keep [%s] not confirmed after [%v] blocks, trying to submit again"` ?